test/bdd/steps/common.mjs

```diff
@@ -266,28 +75,83 @@ Given(
+            logFileStream.write(data);
+        });
+        forkedNode.on('message', async (response) => {
+            if (response.error) {
+                // todo handle error
+            } else {
+                // todo if started
+                const client = new DkgClientHelper({
+                    endpoint: 'http://localhost',
+                    port: 8900,
+                    useSSL: false,
+                    timeout: 25,
+                    loglevel: 'trace',
+                });
+                this.state.bootstraps.push({
+                    client,
+                    forkedNode,
+                    configuration: bootstrapNodeConfiguration,
+                    nodeRpcUrl: `http://localhost:${bootstrapNodeConfiguration.rpcPort}`,
+                });
+            }
+            done();
+        });
+    },
+);
+// TODO: Implement input error handling
+Given(
+    'I setup node {int} with {word} set to {string}',
     { timeout: 120000 },
-    function setupPublishNode(nodeIndex, done) {
-        this.logger.log(`I setup node ${nodeIndex} with invalid configuration`);
-        const wallet = this.state.localBlockchain.getWallets()[nodeIndex];
-        const managementWallet = this.state.localBlockchain.getWallets()[nodeIndex + 27];
-        const rpcPort = 8901 + nodeIndex - 1;
-        const nodeName = `origintrail-test-${nodeIndex - 1}`;
-        const nodeConfiguration = deepExtend(
+    function setupPublishNode(nodeNum, propertyName, propertyValue, done) {
+        const propertyNameSplit = propertyName.split('.');
+        this.logger.log(`I setup node ${nodeNum} with invalid configuration`);
+        expect(
+            Object.prototype.hasOwnProperty.call(defaultConfiguration, propertyNameSplit[0]),
+            `Property ${propertyName} doesn't exist`,
+        ).to.be.equal(true);
+        const nodeIndex = Object.keys(this.state.nodes).length;
+        const wallets = this.state.localBlockchain.getWallets();
+        const wallet = wallets[nodeIndex + 1];
+        const managementWallet =
+            this.state.localBlockchain.getWallets()[nodeIndex + 1 + Math.floor(wallets.length / 2)];
+        const rpcPort = 8901 + nodeIndex;
+        const nodeName = `origintrail-test-${nodeIndex}`;
+        const nodeConfiguration = DeepExtend(
             {},
             defaultConfiguration,
             createNodeConfiguration.call(
                 this,
                 wallet,
                 managementWallet,
-                nodeIndex - 1,
+                nodeIndex,
                 nodeName,
                 rpcPort,
             ),
         );
-        nodeConfiguration.minimumAckResponses.publish = 10;
+        const propertyNameSplitLen = propertyNameSplit.length;
+        let propName = nodeConfiguration;
+        for (let i = 0; i < propertyNameSplitLen - 1; i += 1) {
+            propName = propName[propertyNameSplit[i]];
+        }
+        // TODO: ugly workaround, change it
```

Comment:
Not sure what you are doing here, but we should probably do different implementation. Can you elaborate more

Reply:
I take the parameter propertyName and separate it by dot('.') character and put it in a propertyNameSplit array. Then i go trough configuration until i match the the last propertyName in an array.

For example:
i pass 'modules.autoUpdater.enabled as properyName, and 'true' as propertyValue. argument is separated and stored in propertyNameSplit =['modules', 'autoUpdater', 'enabled'] array. Then the properties are matched in configuration file until i get 'modules.autoUpdater property so i can change the value of 'enabled' field :

first iteration
==========
// matches modules property
propName = 
 {
  blockchain: { ... },
  autoUpdater: {...},
  network{...}, ...
 }
second iteration
============
// matches autoUpdated property
propName = { enabled : false, implementation : {...} }

Loop is finished
==========
// propertyValue is assigned to propertyName
propName['enabled'] = 'true'

In this way different property names and their values can be passed directly to the this step , without the need to define multiple steps for different properties.